sounds/management/commands/analysis_orchestrator.py

```diff
@@ -0,0 +1,96 @@
+#
+# Freesound is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Freesound is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     See AUTHORS file.
+#
+
+import json
+import logging
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from sounds.models import Sound, SoundAnalysis
+
+console_logger = logging.getLogger("console")
+
+# Dictionary that links simple names and versions of the analyzers to their actual docker images
+# Needs to be updated everytime a new analyzer or analyzer version is added
+ANALYZERS_DICT = {
+    "fs-essentia:1": "fs-essentia-extractor:20210525_9a3bd10",
+    "audio-commons:1": "ac-extractor:20210525_9a3bd10",
+    "audioset-vggish:1": "audioset-vggish-extractor:20210615_f44030e"
+}
+
+
+class Command(BaseCommand):
+
+    help = """Checks if there are sounds that have not been analyzed by the analyzers defined in the Analysis 
+    Configuration File (or that are analyzed with older versions) and send jobs to the analysis workers if needed"""
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--analysis_config_file',
+            action='store',
+            required=True,
+            help='Absolute path to the analysis configuration file')
+        parser.add_argument(
+            '--status',
+            action="store_true",
+            help='Flag to print the main status of the analysis. It does not trigger any analysis.')
+
+    def handle(self, *args, **options):
+        # Read config json
+        with open(options['analysis_config_file']) as json_file:
+            config_file = json.load(json_file)
+        
+        # Print information about the already analyzed sounds if the status flag exists
+        if options['status']:
+            n_analyzed = len(SoundAnalysis.objects.all())
+            n_sounds = len(Sound.objects.all())
+            n_analyzers = len(config_file['analyzer:version'])
+            total2analyze = n_sounds * n_analyzers
+            console_logger.info("{0} analysis performed. In total, there can be {1} ({2} sounds x {3} analyzers in config file).".format(n_analyzed, total2analyze, n_sounds, n_analyzers))
+            
+            # Count number of analysis to be performed
+            n_analysis = 0
+            for a in config_file['analyzer:version']:
+                # Check all sounds available
+                for s in Sound.objects.all():
```

Comment:
We will iterate all sounds for every analyzer!
Why not simply using SoundAnalysis.objects.filter(...) for every analyzer to know how many sounds have already been analyzed (also those that failed or skipped) and the ones remaining?

I think what the status command should basically output is a table like:

```
analyzer name        | # ok   | # failed   | # skipped    | # missing
essentia:134678x     | 2459   | 123        | 45           | 364736   
...
```

Running the orchestrator will create jobs for the "missing" sounds for each analyzer. Maybe also add some flags to re-trigger "skipped" and/or "failed" and/or  "ok" sounds. Maybe the sounds to be analyzed could be determined with a flag `--include-skipped` , `--include-failed`, `--include-ok`, `--include-missing` for each column. The usual thing will be to call the orchestrator with `--include-missing` only, but if we want to re-process failed sounds, maybe we use `--include-failed` or a combination of other flags.

Reply:
Because the sounds that haven't been analyzed don't have a SoundAnalysis object... 
If I understand well, the number of missing files for each analyzer would be `len(Sound.objects.all()) - len(SoundAnalysis.objects.filter(analyzer=analyzer, analyzer_version=version))`, do you mean that? 
In any case, right now we are missing a field called `state` in the SoundAnalysis model that is set to "OK", "SK" or "FA" when the analysis finishes, aren't we?

Reply:
yes, we're missing that status field. also correct calculating number of missing sounds by subtracting the ones done from the total. however, the way you do it will take long. you should use counts: `Sound.objects.all().count() - SoundAnalysis.objects.filter(analyzer=analyzer, analyzer_version=version).count()` (and also avoid repeating queries, the first one you can keep it in a variable for sure so you don't count for every analyzer)